### PR TITLE
Add support for partition cache verification

### PR DESCRIPTION
**Pull request by NikhilCollooru**
```
== RELEASE NOTES ==

Hive Changes
* Add support for partition cache validation. This can be enabled by setting `hive.partition-cache-validation-percentage` configuration parameter.
```

### CODE REVIEW

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/MetastoreClientConfig.java [hunk 1]
```diff
@@ -46,6 +46,7 @@
     private boolean partitionVersioningEnabled;
     private MetastoreCacheScope metastoreCacheScope = MetastoreCacheScope.ALL;
     private boolean metastoreImpersonationEnabled;
+    private boolean partitionCacheVerificationEnabled;
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/MetastoreClientConfig.java:49]: Shall we make it a 0 - 100 integer to sample how much traffic we do validation? Check `orcWriterValidationPercentage` as an example

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java [hunk 2]
```diff
@@ -605,7 +614,12 @@ protected void invalidateTable(String databaseName, String tableName)
     @Override
     public Optional<Partition> getPartition(MetastoreContext metastoreContext, String databaseName, String tableName, List<String> partitionValues)
     {
-        return get(partitionCache, getCachingKey(metastoreContext, hivePartitionName(databaseName, tableName, partitionValues)));
+        KeyAndContext<HivePartitionName> key = getCachingKey(metastoreContext, hivePartitionName(databaseName, tableName, partitionValues));
+        Optional<Partition> result = get(partitionCache, key);
+        if (partitionCacheVerificationEnabled) {
```
Comment (highker) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java:619]: we can check if the percentage is 0; if it's > 0, we can make a random number to decide if we wanna do validation or not.

**Review by highker — COMMENTED** [review]
one minor comments; will leave it to @shixuan-fan for final review

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java [hunk 3]
```diff
@@ -605,7 +614,12 @@ protected void invalidateTable(String databaseName, String tableName)
     @Override
     public Optional<Partition> getPartition(MetastoreContext metastoreContext, String databaseName, String tableName, List<String> partitionValues)
     {
-        return get(partitionCache, getCachingKey(metastoreContext, hivePartitionName(databaseName, tableName, partitionValues)));
+        KeyAndContext<HivePartitionName> key = getCachingKey(metastoreContext, hivePartitionName(databaseName, tableName, partitionValues));
+        Optional<Partition> result = get(partitionCache, key);
```
Comment (shixuan-fan) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java:618]: It seems that it is possible we are comparing metastore against metastore in this case, which is likely fine but just to call it out

**Review by shixuan-fan — APPROVED** [review]
Looks good. I don't have a strong opinion on percentage, but it does not seem hurt to add.

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/MetastoreClientConfig.java (hunk 1 above)
Comment (NikhilCollooru) [reply presto-hive-metastore/src/main/java/com/facebook/presto/hive/MetastoreClientConfig.java]: Done.

**Pull request merged**